### DISCUSSION TITLE
Installing home-brew on macOS Sonoma

### DISCUSSION THREAD

**Discussion by ImNoName3138** [Homebrew/discussions#4727] [Getting started] [answered]
### Output of `brew config`

```shell
Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /bin/chmod u+rwx /usr/local/bin /usr/local/sbin /usr/local/share /usr/local/share/man /usr/local/share/man/man8
==> /usr/bin/sudo /bin/chmod g+rwx /usr/local/bin /usr/local/sbin /usr/local/share /usr/local/share/man /usr/local/share/man/man8
==> /usr/bin/sudo /usr/sbin/chown Minhson\ :3 /usr/local/bin /usr/local/sbin /usr/local/share /usr/local/share/man /usr/local/share/man/man8
chown: Minhson : illegal user name
Failed during: /usr/bin/sudo /usr/sbin/chown Minhson\ :3 /usr/local/bin /usr/local/sbin /usr/local/share /usr/local/share/man /usr/local/share/man/man8
Minhson :3@Nonames-MacBook-Air-M2 ~ % xcode-select --install
xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates
```

### Output of `brew doctor`

```shell
s
```

### Description of issue

I can't install home-brew on macOS Sonoma for apple porting tool kit

**Comment by SMillerDev** [comment]
MacOS Sonoma is not officially supported yet. Any issues you encounter should be fixed by you. In this case it seems like you messed with `$USER` though, so it would not work with other OS versions either.

**Reply by ImNoName3138** [reply]
I tried to change my macOS username before should that be the reason bc ive seen a lot of videos of people installing home-brew on macOS Sonoma for game porting toolkit

**Reply by ZhongRuoyu** [reply]
Did you somehow modify the environment variable `USER` in one of your shell configs (`~/.zprofile`, etc.)? If so, try

```bash
export USER="$(id -un)"
```

before installing.

NB. Changing `USER` can lead to many unintended consequences and is not recommended.

**Reply by ImNoName3138** [reply]
I think I tired modifying the username by creating another user and GO to that user to change the username/file name

**Reply by ImNoName3138** [reply]
bcuz the user profile name on Mac is completely different from the profile file name

**Reply by ZhongRuoyu** [reply]
> I think I tired modifying the username by creating another user and GO to that user to change the username/file name

What did you do, exactly?

> bcuz the user profile name on Mac is completely different from the profile file name

What were you referring to by "profile file name"? What are these on your Mac, respectively?

**Reply by ImNoName3138** [reply]
So like I bought a used Mac but when I changed the Apple ID on the Mac
the user profile still used the old name

**Reply by ImNoName3138** [reply]
so I create a new user and go to the new user and change the user I'm currently using in the settings but the user file name still the old name

**Reply by ImNoName3138** [reply]
also I'm wondering about the part that says illegal username

**Reply by ZhongRuoyu** [reply]
If you followed the steps in https://support.apple.com/en-us/HT201548, then it should be fine.

However, what matters here is what your current username is. Can you show the output of:

```
id -un
```

**Reply by ImNoName3138** [reply]
its 
Minhson :3@Nonames-MacBook-Air-M2 ~ % id -un
Minhson :3

**Reply by ZhongRuoyu** [reply]
Many unexpected errors can occur when the username contains characters like spaces and colons. I strongly recommend changing your username to something like `Minhson` instead (following steps in https://support.apple.com/en-us/HT201548). The Homebrew installer might be the first that throws a weird error at you, but it's most likely not the last.

**Reply by ImNoName3138** [reply]
im not sure why but the user folder name isnt same to the command

**Reply by ZhongRuoyu** [reply, marked as answer]
The name of your home directory does not necessarily need to be the same as your username. (Though it's usually the case.) As shown in your command output, your current username is `Minhson :3`.

To change your username (and rename your home directory to match the username), please carefully follow the steps in https://support.apple.com/en-us/HT201548.

**Reply by ImNoName3138** [reply]
i tried changing the file name and the user file directory and it works tysm!
